### PR TITLE
Design fixes for row actions

### PR DESCRIPTION
**Pull request by nambatee**
Follow-up on Carolyn's comments in https://phabricator.wikimedia.org/T169921

### CODE REVIEW

File: Wikipedia/Code/WMFHistoryTableViewController.m [hunk 1]
```diff
@@ -129,17 +129,26 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
-
-    UITableViewRowAction *saveAction = [self saveAction:indexPath];
-    saveAction.backgroundColor = [UIColor wmf_blue];
+    shareAction.backgroundColor = self.theme.colors.secondaryAction;
+    
+    NSMutableArray<UITableViewRowAction *> *actions = [[NSMutableArray alloc] initWithObjects: deleteAction, shareAction, nil];
+    
+    if ([[self savedPageList] isSaved:[self urlAtIndexPath:indexPath]]) {
+        UITableViewRowAction *unsaveAction = [self unsaveAction:indexPath];
+        unsaveAction.backgroundColor = [UIColor wmf_blue];
```
Comment (joewalsh) [inline Wikipedia/Code/WMFHistoryTableViewController.m]: Should be `self.theme.colors.link`

File: Wikipedia/Code/WMFHistoryTableViewController.m [hunk 2]
```diff
@@ -129,17 +129,26 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
+        [actions addObject:unsaveAction];
+    } else {
+        UITableViewRowAction *saveAction = [self saveAction:indexPath];
+        //TODO: should it be using a theme color?
+        saveAction.backgroundColor = [UIColor wmf_blue];
```
Comment (joewalsh) [inline Wikipedia/Code/WMFHistoryTableViewController.m]: ^ same comment

File: Wikipedia/Code/WMFSavedArticleTableViewController.m [hunk 3]
```diff
@@ -86,12 +86,12 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 
 -(NSArray *)tableView:(UITableView *)tableView editActionsForRowAtIndexPath:(NSIndexPath *)indexPath {
     
-    UITableViewRowAction *deleteAction = [self deleteAction:indexPath];
-    deleteAction.backgroundColor = self.theme.colors.destructive;
+    UITableViewRowAction *unsaveAction = [self unsaveAction:indexPath];
+    unsaveAction.backgroundColor = [UIColor wmf_blue];
```
Comment (joewalsh) [inline Wikipedia/Code/WMFSavedArticleTableViewController.m]: ^ same comment (should be link color)

File: Wikipedia/Code/WMFArticleListTableViewController.m [hunk 4]
```diff
@@ -284,6 +284,10 @@ - (NSString *)saveActionText {
     return WMFLocalizedStringWithDefaultValue(@"article-save", nil, nil, @"Save", @"Text of the article list row action shown on swipe which allows the user to save the article");
 }
 
+- (NSString *)unsaveActionText {
+    return WMFLocalizedStringWithDefaultValue(@"article-unsave", nil, nil, @"Unsave", @"Text of the article list row action shown on swipe which allows the user to unsave the article");
```
Comment (joewalsh) [inline Wikipedia/Code/WMFArticleListTableViewController.m]: This should use `[WMFSaveButton shortUnsaveTitle]`

File: Wikipedia/Code/WMFArticleListTableViewController.m [hunk 5]
```diff
@@ -284,6 +284,10 @@ - (NSString *)saveActionText {
     return WMFLocalizedStringWithDefaultValue(@"article-save", nil, nil, @"Save", @"Text of the article list row action shown on swipe which allows the user to save the article");
```
Comment (joewalsh) [inline Wikipedia/Code/WMFArticleListTableViewController.m]: This should use `[WMFSaveButton shortSaveTitle]`

Comment (joewalsh) [inline Wikipedia/Code/WMFArticleListTableViewController.m]: Added a ticket to track this issue: https://phabricator.wikimedia.org/T170938 - doesn't necessarily need to happen with this PR.

Comment (nambatee) [inline Wikipedia/Code/WMFArticleListTableViewController.m]: 👍 thanks! I'll grab it when I'm finished with theme controls in Settings

**Pull request merged**